tests/core/channels/test_audiocodes.py

```diff
@@ -0,0 +1,141 @@
+import pytest
+from rasa.core import utils
+
+
+def handle_activities_test_cb(expectedString):
+    async def cb(resultUM):
+        assert expectedString == resultUM.text
+
+    return cb
+
+
+def handle_activities_test_throw_cb(expectedString):
+    def cb(resultUM):
+        raise Exception("Bad result")
+
+    return cb
+
+
+def test_audiocodes_channel():
+
+    from rasa.core.channels.audiocodes import AudiocodesInput
+    import rasa.core
+
+    input_channel = AudiocodesInput("some_generated_token")
+
+    s = rasa.core.run.configure_app([input_channel], port=5004)
+
+    routes_list = utils.list_routes(s)
+    assert routes_list.get("ac_webhook.health", "").startswith("/webhooks/audiocodes")
+    assert routes_list.get("ac_webhook.receive", "").startswith(
+        "/webhooks/audiocodes/webhook"
+    )
+
+
+@pytest.mark.asyncio
+async def test_audiocodes_handle_events():
+
+    from rasa.core.channels.audiocodes import AudiocodesInput, AudiocodesOutput
+
+    ac_output = AudiocodesOutput()
+    conversation_id = "some-generated-uuid"
+
+    async def test_message(activity, expected_text, cb=handle_activities_test_cb):
+        message = {"conversation": conversation_id, "activities": [activity]}
+        return await c.handleActivities(message, ac_output, cb(expected_text))
+
+    input_channel = AudiocodesInput("some_generated_token")
+    c = AudiocodesInput.Conversation(conversation_id)
+    input_channel.conversations[conversation_id] = c
+
+    TEXT_MESSAGE = {
+        "type": "message",
+        "text": "This is a text message",
+        "parameters": {"confidence": 0.8172},
+        "id": "generated id 1",
+    }
+
+    await test_message(TEXT_MESSAGE, "This is a text message")
+
+    TEXT_MESSAGE["id"] = "generated id 1.1"
+    await test_message(
+        TEXT_MESSAGE, "This is a text message", cb=handle_activities_test_throw_cb
+    )
+    assert len(ac_output.messages) == 1
+    assert ac_output.messages[0]["name"] == "hangup"
+
+    START_EVENT = {
+        "type": "event",
+        "name": "start",
+        "parameters": {
+            "caller": "some_caller",
+            "callee": "123456789",
+            "callerHost": "ip",
+            "calleeHost": "other_ip",
+        },
+        "id": "generated id 2",
+    }
+
+    await test_message(
+        START_EVENT,
+        "/start{'caller': 'some_caller', 'callee': '123456789', 'callerHost': 'ip', 'calleeHost': 'other_ip'}",
+    )
+
+    DTMF_EVENT = {"type": "event", "name": "DTMF", "value": 3, "id": "generated id 3"}
+
+    await test_message(DTMF_EVENT, "/DTMF{'value': 3}")
+
+    def raise_if_called(p):
+        def cb():
+            assert False, "Shouldn't be called"
+
+        return cb
```

Comment:
Can this replace `handle_activities_test_throw_cb`? I don't see a reason to have both.